extensions/sd_api_pictures/script.py

```diff
@@ -86,14 +88,37 @@ def triggers_are_in(string):
     # regex searches for send|main|message|me (at the end of the word) followed by
     # a whole word of image|pic|picture|photo|snap|snapshot|selfie|meme(s),
     # (?aims) are regex parser flags
-    return bool(re.search('(?aims)(send|mail|message|me)\\b.+?\\b(image|pic(ture)?|photo|snap(shot)?|selfie|meme)s?\\b', string))
+    return bool(re.search('(?aims)(send|mail|message|me)\\b.+?\\b(image|pic(ture)?|photo|snap(shot)?|selfie|meme|painting)s?\\b', string))
 
 
 def input_modifier(string):
     """
     This function is applied to your text inputs before
     they are fed into the model.
     """
+    # Because it is impossible to get 'name2' unless this is called by server.py, this little workaround will get the name of the bot
+    # I mostly stole this from chat.load_character(), but getting the proper folder name has the same sort of problem as 'name2'
+    if shared.character != 'None':
+        found_file=False
+        folder1 = 'characters'
+        folder2 = 'characters/instruction-following'
+        for folder in [folder1, folder2]:
+            for extension in ["yml", "yaml", "json"]:
+                filepath = Path(f'{folder}/{shared.character}.{extension}')
+                if filepath.exists():
+                    found_file = True
+                    break
+            if found_file:
+                break
+        file_contents = open(filepath, 'r', encoding='utf-8').read()
+        data = json.loads(file_contents) if extension == "json" else yaml.safe_load(file_contents)
+    
+        if 'your_name' in data and data['your_name'] != '':
+            name1 = data['your_name']
+        name2 = data['name'] if 'name' in data else (data['char_name'] if 'char_name' in data else "Assistant")
+    else:
+        name2 = "Assistant"
+
```

Comment:
I like the workaround to get name1 and name2 @lines99-121 but I feel like it should be factored out into separate function.

After all, this does mostly unrelated things to input_modifier and pulls info from the filesystem, solving a problem which ideally shouldn't even exist

Reply:
Also, how do you get these code snippets inline with the rest of the comments?